extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/MSQCompactionRunner.java

```diff
@@ -237,7 +238,11 @@ private static DataSourceMSQDestination buildMSQDestination(
         dataSchema.getDataSource(),
         dataSchema.getGranularitySpec().getSegmentGranularity(),
         null,
-        ImmutableList.of(replaceInterval)
+        ImmutableList.of(replaceInterval),
+        dataSchema.getDimensionsSpec()
```

Comment:
What about cases where dimension schema is not present in the compaction spec, would those dimensions be present in this schema ?

Reply:
Yes, in those cases the segment schemas of compaction candidates are analyzed and a coerced schema is used.